client/src/components/ClickOutComponent/ClickOutComponent.js

```diff
@@ -0,0 +1,76 @@
+/* global document */
+import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+
+/**
+ * Wrapper component that fires a handler (`onClickOut`) when a user clicks
+ * outside of the component.
+ *
+ * Example:
+ *
+ * ```jsx
+ *    class TestComponent extends React.Component {
+ *      ...
+ *
+ *      handleClickOut() {
+ *        alert('User clicked outside of TestComponent.');
+ *      }
+ *
+ *      render() {
+ *        return (
+ *          <ClickOutComponent onClickOut={this.handleClickOut} callerComponent={this}>
+ *            <div>... </div>
+ *          </ClickOutComponent>
+ *        );
+ *       }
+ *    < /ClickOutComponent>
+ *  ```
+ */
+
+class ClickOutComponent extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.wasClicked = false;
```

Comment:
have you tried move this to `this.state` and use `this.setState`, I can see why it potentially could be undesired, but might be worth using.